### ISSUE TITLE
Feature: warn about possibly failing pipelines when set -e is set without -o pipefail

### ISSUE THREAD

**Issue by alexjurkiewicz** [koalaman/shellcheck#1022]
I think this is a pretty common gotcha

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash -e
myvar=$(false | grep foo)
export myvar
```

#### Here's what I wanted or expected to see:
```
myvar=$(false | grep foo)
        ^-- SCXXXX: Warning: You have enabled `set -e` but this command can silently fail. Split the command or use `set -o pipefail`
```
